php/libraries/Recording.class.inc

```diff
@@ -0,0 +1,425 @@
+     * Return a SPLFileInfo object based on this electrodes' file's properties
+     *
+     * @return \SplFileInfo
+     */
+    public function getElectrodesFileInfo(): \SplFileInfo
+    {
+        $imagepath = \NDB_Factory::singleton()
+            ->config()
+            ->getSetting('imagePath');
+
+        $fullpath = $imagepath . $this->_electrode_file_path;
+
+        $info = new \SplFileInfo($fullpath);
+
+        return $info;
+    }
+
+    /**
+     * Return a SPLFileInfo object based on this events.class' file's properties
+     *
+     * @return \SplFileInfo
+     */
+    public function getEventsFileInfo(): \SplFileInfo
+    {
+        $imagepath = \NDB_Factory::singleton()
+            ->config()
+            ->getSetting('imagePath');
+
+        $fullpath = $imagepath . $this->_event_file_path;
+
+        $info = new \SplFileInfo($fullpath);
+
+        return $info;
+    }
+
+    /**
+     * Return a SPLFileInfo object based on this archive's properties
+     *
+     * @return \SplFileInfo
+     */
+    public function getArchiveFileInfo(): \SplFileInfo
+    {
+        $imagepath = \NDB_Factory::singleton()
+            ->config()
+            ->getSetting('imagePath');
+
+        $fullpath = $imagepath . $this->_archive_file_path;
+
+        $info = new \SplFileInfo($fullpath);
+
+        return $info;
+    }
+
+    /**
+     * Get all channels info from the physiological_channel table for that
+     * Recording.
+     *
+     * @return \LORIS\RecordingChannels[]
+     */
+    public function getChannels(): array
+    {
+        $dbrows = \NDB_Factory::singleton()
+            ->database()
+            ->pselect(
+                '
+                       SELECT
+                         pc.Name                as channelname,
+                         pc.Description         as channeldescription,
+                         pct.ChannelTypeName    as channeltype,
+                         pct.ChannelDescription as channeltypedescription,
+                         pst.ChannelStatus      as channelstatus,
+                         pc.StatusDescription   as statusdescription,
+                         pc.SamplingFrequency   as samplingfrequency,
+                         pc.LowCutoff           as lowcutoff,
+                         pc.HighCutoff          as highcutoff,
+                         pc.ManualFlag          as manualflag,
+                         pc.Notch               as notch,
+                         pc.Reference           as reference,
+                         pc.Unit                as unit,
+                         pc.FilePath            as filepath
+                       FROM
+                         physiological_channel pc 
+                       JOIN physiological_channel_type pct
+                         USING (PhysiologicalChannelTypeID)
+                       JOIN physiological_status_type pst
+                         USING (PhysiologicalStatusTypeID)
+                       JOIN physiological_file f
```

Comment:
Is this join necessary? No parameter from this table as selected and parameter used in `WHERE` is available in other table.

Reply:
right. Removed the extra unnecessary JOIN